documentation/tutorials/get-started-with-ash-and-phoenix.md

```diff
@@ -0,0 +1,481 @@
+
+```elixir
+  ...
+
+  ### READ ACTION - read blog post(s) ###
+  # we need first_post for update action later
+    assert [first_post = %{title: "hello world"}] =
+             MyAshPhoenixApp.Blog.read!(MyAshPhoenixApp.Blog.Post)
+
+    # will return:
+    #
+    # [
+    #   #MyAshPhoenixApp.Blog.Post<
+    #     __meta__: #Ecto.Schema.Metadata<:loaded, "posts">,
+    #     id: "d70dd979-0b30-4a3f-beb2-2d5bb2e24af7",
+    #     title: "hello world",
+    #     content: nil,
+    #     aggregates: %{},
+    #     calculations: %{},
+    #     __order__: nil,
+    #     ...
+    #   >
+    # ]
+
+    ...
+```
+
+Our post doesn't have any contents. Lets put some text in the `:content` attribute.
+For this we will use an `:update` action:
+
+```elixir
+  ...
+
+  ### UPDATE ACTION - update existing blog post ###
+    # notice how you have to parse in an existing resource to the changeset
+    assert %{
+             title: "hello world",
+             content: "hello to you too!"
+           } =
+             Ash.Changeset.for_update(first_post, :update, %{content: "hello to you too!"})
+             |> MyAshPhoenixApp.Blog.update!()
+
+    # will return:
+    #
+    # #MyAshPhoenixApp.Blog.Post<
+    #   __meta__: #Ecto.Schema.Metadata<:loaded, "posts">,
+    #   id: "d70dd979-0b30-4a3f-beb2-2d5bb2e24af7",
+    #   title: "hello world",
+    #   content: "hello to you too!",
+    #   aggregates: %{},
+    #   calculations: %{},
+    #   __order__: nil,
+    #   ...
+    # >
+
+  ...
+```
+
+Finally lets delete the post, and check its been removed from the resource, by performing one last read actions.
+
+```elixir
+  ...
+
+  ### DELETE ACTION - delete existing blog post ###
+    assert :ok ==
+             Ash.Changeset.for_destroy(first_post, :destroy)
+             |> MyAshPhoenixApp.Blog.destroy!()
+
+    # verifying no rows in resource
+    assert [] == MyAshPhoenixApp.Blog.read!(MyAshPhoenixApp.Blog.Post)
+  end
+end
+```
+
+Now lets run the test.
+
+```bash
+> mix test test/my_ash_phoenix_app/blog/resources/post_test.exs
+
+.
+Finished in 0.2 seconds (0.2s async, 0.00s sync)
+1 test, 0 failures
+
+Randomized with seed 333442
+```
+
+It works 🎉🥳!
+
+### But how to integrate with Phoenix?
+
+Its simple, you can call the code in the tests above in your controller or LiveView.
+Here are some simple examples.
+
+In a controller:
+
+```elixir
+  def index(conn, _params) do
+    conn
+    |> assign(:posts, MyAshPhoenixApp.Blog.read!(MyAshPhoenixApp.Blog.Post))
+    |> render("index.html")
+  end
+```
+
+In a LiveView:
+
+```elixir
+  def mount(_params, _session, socket) do
+    socket =
+      assigns(socket, :posts, MyAshPhoenixApp.Blog.read!(MyAshPhoenixApp.Blog.Post))
+
+    {:ok, socket}
+  end
+```
+
+There are more idiomatic ways to interact with ash in the view layer, and we'll cover them. But this will do for now.
```

Comment:
Ash should always be capitalized.

Reply:
addressed: b426d8893755af97cef6436eb6fba1160cef80ba